src/lib.rs

```diff
@@ -59,4 +63,57 @@ impl Calc {
             .collect::<Vec<String>>();
         return msgs.join("\n");
     }
+
+    pub fn eval(opcode: AstNode) -> Result<u64, String> {
+        let bytecode = &mut vec![];
+        Self::to_bytecode(opcode, bytecode);
```

Comment:
I'd say the call to `to_bytecode()` should be outside of `eval()`, but it doesn't matter too much.

Reply:
I think my naming might be confusing, there's `eval_bytecode` function that can be used separately

Reply:
What I mean is that I'd consider evaluating the bytecode separate from constructing it.

Reply:
I thought I did with the split or eval and eval_bytecode. Anyway make things clearer I separated and left only one function  - `eval` :point_right: https://github.com/softdevteam/pavel.yaiwr/pull/10/commits/8e960eada019da4a18def6435a73c1fa200a2f0e